src/Compilers/CSharp/Portable/Binder/Semantics/AccessCheck.cs

```diff
@@ -100,6 +100,12 @@ internal static class AccessCheck
                 case SymbolKind.NamedType:
                     return IsNamedTypeAccessible((NamedTypeSymbol)symbol, within, ref useSiteDiagnostics, basesBeingResolved);
 
+                case SymbolKind.Alias:
+                    return IsSymbolAccessibleCore(((AliasSymbol)symbol).Target, within, null, out failedThroughTypeCheck, compilation, ref useSiteDiagnostics, basesBeingResolved);
+
+                case SymbolKind.Discard:
+                    return IsSymbolAccessibleCore(((DiscardSymbol)symbol).Type, within, null, out failedThroughTypeCheck, compilation, ref useSiteDiagnostics, basesBeingResolved);
```

Comment:
>((DiscardSymbol)symbol).Type [](start = 50, length = 28)

Is ```((DiscardSymbol)symbol).Type``` guaranteed to be not null in all scenarios?

Reply:
> Is `((DiscardSymbol)symbol).Type` guaranteed to be not null in all scenarios?

The compilers do not expose discard symbol with null types.

---
In reply to: [192574898](https://github.com/dotnet/roslyn/pull/27312#discussion_r192574898) [](ancestors = 192574898)